worker/apiconfigwatcher/manifold.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package apiconfigwatcher
+
+import (
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/voyeur"
+	"launchpad.net/tomb"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/worker"
+	"github.com/juju/juju/worker/dependency"
+)
+
+var logger = loggo.GetLogger("juju.worker.apiconfigwatcher")
+
+type ManifoldConfig struct {
+	AgentName          string
+	AgentConfigChanged *voyeur.Value
+}
+
+// Manifold returns a dependency.Manifold which wraps the an agent's
+// voyeur.Value is set whenever it the agent config is
+// changed. Whenever the API server addresses in the config change the
+// manifold will bounce itself.
+//
+// The manifold is intended to be a dependency for the api-caller
```

Comment:
I had the same concern. Some better solutions might be to have the worker bounce only when *all* the addresses change or if the API address that the agent is using goes away.

Regarding the former I'l worried that there might be some addresses in private address ranges being advertised that remain the same between controllers. 

The problem with the latter is that the apiconfigwatcher has no idea which address is in use by the api-caller.

For now I think we live with some extra bouncing. I'm creating a card now to track this.